apollo/lib/router.py

```diff
@@ -63,10 +63,16 @@ def _http_method(self, func, http_method, *outer_args, permission='public',
                      **outer_kwargs):
         route = getattr(super(SecureRouter, self), http_method)
 
-        @route(*outer_args, **outer_kwargs)
-        def wrapped(request: Request, *args, **kwargs):
-            self.acl_policy.validate_permission(permission, request)
-            return func(*args, **kwargs)
+        if inspect.iscoroutinefunction(func):
```

Comment:
Router tests not updated to include this new branch. I'm also not entirely sure yet if this is the best way but that's beside the point.